qualtran/_infra/gate_with_registers.py

```diff
@@ -308,17 +308,22 @@ def decompose_from_registers(
     def _decompose_with_context_(
         self, qubits: Sequence[cirq.Qid], context: Optional[cirq.DecompositionContext] = None
     ) -> cirq.OP_TREE:
+        from qualtran.cirq_interop._cirq_to_bloq import InteropQubitManager
+
         quregs = split_qubits(self.signature, qubits)
         if context is None:
             context = cirq.DecompositionContext(cirq.ops.SimpleQubitManager())
         try:
             return self.decompose_from_registers(context=context, **quregs)
-        except DecomposeNotImplementedError as e:
+        except DecomposeNotImplementedError:
             pass
         try:
-            qm = context.qubit_manager
             cbloq = self.decompose_bloq()
-            circuit, out_quregs = cbloq.to_cirq_circuit(qubit_manager=qm, **quregs)
+            in_quregs = {reg.name: quregs[reg.name] for reg in self.signature.lefts()}
+            # Input qubits can get de-allocated by cbloq.to_cirq_circuit, thus mark them as managed.
```

Comment:
@mpharrigan this was the issue

Reply:
oh, so it was just an accounting issue rather than the accounting finding an actual error?

Reply:
Yes, that's correct. I've added tests so the PR is ready for review now.